probe.go

```diff
@@ -1011,6 +1012,144 @@ func probeCertificates(c FortiHTTP) ([]prometheus.Metric, bool) {
+		)
+		realServerBytesProcessed = prometheus.NewDesc(
+			"fortigate_lb_real_server_processed_bytes_total",
+			"Number of bytes processed by this real server",
+			[]string{"vdom", "virtual_server", "id"}, nil,
+		)
+	)
+
+	type RealServer struct {
+		IP     string `json:"real_server_ip"`
+		Port   int    `json:"real_server_port"`
+		ID     int    `json:"real_server_id"`
+		Mode   string `json:"mode"`
+		Status string `json:"status"`
+		// MonitorEvents  int    `json:"monitor_events"`
+		ActiveSessions float64 `json:"active_sessions"`
+		RTT            string  `json:"RTT"`
+		BytesProcessed float64 `json:"bytes_processed"`
+	}
+
+	type VirtualServer struct {
+		Name        string       `json:"virtual_server_name"`
+		IP          string       `json:"virtual_server_ip"`
+		Port        int          `json:"virtual_server_port"`
+		Type        string       `json:"virtual_server_type"`
+		RealServers []RealServer `json:"list"`
+	}
+
+	type LoadBalanceResponse struct {
+		HTTPMethod string          `json:"http_method"`
+		Results    []VirtualServer `json:"results"`
+		VDOM       string          `json:"vdom"`
+		Path       string          `json:"path"`
+		Name       string          `json:"name"`
+		Status     string          `json:"status"`
+		Serial     string          `json:"serial"`
+		Version    string          `json:"version"`
+		Build      int             `json:"build"`
+	}
+
+	// Consider implementing pagination to remove this limit of 1000 entries
+	var rs []LoadBalanceResponse
+	if err := c.Get("api/v2/monitor/firewall/load-balance", "vdom=*&start=0&count=1000", &rs); err != nil {
+		log.Printf("Error: %v", err)
+		return nil, false
+	}
+
+	m := []prometheus.Metric{}
+
+	for _, r := range rs {
+		for _, virtualServer := range r.Results {
+			m = append(m, prometheus.MustNewConstMetric(virtualServerInfo, prometheus.GaugeValue, 1, r.VDOM, virtualServer.Name, virtualServer.IP, strconv.Itoa(virtualServer.Port), virtualServer.Type))
+
+			for _, realServer := range virtualServer.RealServers {
+				realServerModeActive, realServerModeStandby, realServerModeDisabled := 0.0, 0.0, 0.0
+				switch realServer.Mode {
+				case "active":
+					realServerModeActive = 1.0
+					break
+				case "standby":
+					realServerModeStandby = 1.0
+					break
+				case "disabled":
+					realServerModeDisabled = 1.0
+					break
+				}
+
+				realServerStatusUp, realServerStatusDown, realServerStatusUnknown := 0.0, 0.0, 0.0
+				switch realServer.Status {
+				case "up":
+					realServerStatusUp = 1.0
+					break
+				case "down":
+					realServerStatusDown = 1.0
+					break
+				default:
+					realServerStatusUnknown = 1.0
+				}
+
+				realServerRTTValue := -1.0
+				if "<1" == realServer.RTT {
+					realServerRTTValue = 1.0
+				} else {
+					var err error
+					realServerRTTValue, err = strconv.ParseFloat(realServer.RTT, 64)
+					if err != nil {
+						log.Printf("Failed to parse RTT value: %v", err)
+						realServerRTTValue = -1.0
```

Comment:
Please only export the metric if there is no error.     
@sboschman or what have been your thoughts regarding this?

Reply:
I was thinking it might be handy to tell something is wrong by the metric value, prompting you to have a look in the logs. Instead of having a sporadic missing metric. Assume if the RTT is exactly 1 sec, it isn't reported as 1000 but as 1s.

On the other hand, a (sporadic) missing metric should perhaps be enough to prompt an inspection of the logs.

Reply:
Ah, I have overlooked that one, the time values should definitely be `seconds` . https://prometheus.io/docs/practices/naming/#base-units     

@bluecmd  WDYT regarding the dropping of the metric.  
ATM we have three options: 
- set the value `-1`, like it is implemented right now
- dropping the metric in the failure case
- dropping the metric in the failure case and return a non success

Reply:
I'd say let's use NaN instead for the error case. It works better using things like `sum` and `avg` etc and will not be lost as noise when people are looking at the metrics in aggregate (as they likely will).

Dropping the metric has the issue that Prometheus does stale metric caching, so it will effectively retain the older value for queries upon it. A NaN will look very odd and cause people to dig deeper, I think.

Reply:
And yes, please expose it as _seconds as per the Prometheus base units as @secustor said.

Reply:
NaN :+1: